File: .skills/openclaw-skills/skills/kesslerio/dialpad/send_sms.py
#!/usr/bin/env python3
"""
Send SMS via Dialpad API.

Usage:
    python3 send_sms.py --to "+14155551234" --message "Hello!"
    python3 send_sms.py --to "+14155551234" "+14155555678" --message "Group message"
    python3 send_sms.py --to "+14155551234" --message "Test" --from "+14155550000"
"""

import argparse
import json
import os
import sys
import urllib.request
import urllib.error


# Configuration
DIALPAD_API_KEY = os.environ.get("DIALPAD_API_KEY")
DIALPAD_API_BASE = "https://dialpad.com/api/v2"


def send_sms(to_numbers, message, from_number=None, infer_country_code=False):
    """
    Send SMS via Dialpad API.

    Args:
        to_numbers: List of E.164 phone numbers (max 10)
        message: SMS text content
        from_number: Optional sender number (must be assigned to your Dialpad account)
        infer_country_code: If True, assume numbers are from sender's country

    Returns:
        dict: API response with SMS details
    """
    if not DIALPAD_API_KEY:
        raise ValueError("DIALPAD_API_KEY environment variable not set")

    if not to_numbers:
        raise ValueError("At least one recipient number is required")

    if len(to_numbers) > 10:
        raise ValueError("Maximum 10 recipients per SMS request")

    url = f"{DIALPAD_API_BASE}/sms"

    payload = {
        "to_numbers": to_numbers,
        "text": message,
        "infer_country_code": infer_country_code
    }

    if from_number:
        payload["from_number"] = from_number

    data = json.dumps(payload).encode("utf-8")

    headers = {
        "Authorization": f"Bearer {DIALPAD_API_KEY}",
        "Content-Type": "application/json",
        "Accept": "application/json"
    }

    request = urllib.request.Request(url, data=data, headers=headers, method="POST")

    try:
        with urllib.request.urlopen(request) as response:
            response_data = response.read().decode("utf-8")
            return json.loads(response_data)
    except urllib.error.HTTPError as e:
        error_body = e.read().decode("utf-8") if e.fp else ""
        try:
            error_data = json.loads(error_body)
            error_msg = error_data.get("error", {}).get("message", error_body)
        except:
            error_msg = error_body or str(e)
        raise RuntimeError(f"Dialpad API error (HTTP {e.code}): {error_msg}")
    except urllib.error.URLError as e:
        raise RuntimeError(f"Network error: {e.reason}")


def main():
    parser = argparse.ArgumentParser(
        description="Send SMS via Dialpad API"
    )
    parser.add_argument(
        "--to",
        nargs="+",
        required=True,
        help="Recipient phone number(s) in E.164 format (e.g., +14155551234)"
    )
    parser.add_argument(
        "--message",
        required=True,
        help="SMS message content"
    )
    parser.add_argument(
        "--from",
        dest="from_number",
        help="Sender phone number (optional, must be assigned to your Dialpad account)"
    )
    parser.add_argument(
        "--infer-country-code",
        action="store_true",
        help="Assume recipient numbers are from sender's country"
    )
    parser.add_argument(
        "--json",
        action="store_true",
        help="Output result as JSON"
    )

    args = parser.parse_args()

    try:
        result = send_sms(
            to_numbers=args.to,
            message=args.message,
            from_number=args.from_number,
            infer_country_code=args.infer_country_code
        )

        if args.json:
            print(json.dumps(result, indent=2))
        else:
            print("SMS sent successfully!")
            print(f"   ID: {result.get('id')}")
            print(f"   Status: {result.get('status')}")
            print(f"   To: {', '.join(result.get('to_numbers', []))}")

        sys.exit(0)

    except ValueError as e:
        print(f"Configuration error: {e}", file=sys.stderr)
        sys.exit(1)
    except RuntimeError as e:
        print(f"API error: {e}", file=sys.stderr)
        sys.exit(2)


if __name__ == "__main__":
    main()

File: .skills/openclaw-skills/skills/teltel-call-center/teltel-send-sms-text-message/scripts/send_sms.js
#!/usr/bin/env node
/**
 * TelTel SMS sender
 *
 * Usage:
 *   node send_sms.js --to "+3712xxxxxxx" --message "Hi" --from "37163030348" [--callback "https://..."] [--apikey "..."] [--dry-run]
 *
 * Env:
 *   TELTEL_API_KEY (preferred)
 *   TELTEL_SMS_FROM (optional default)
 */

const BASE_URL = process.env.TELTEL_BASE_URL || 'https://api.teltel.io/v2';

function parseArgs(argv) {
  const out = {};
  for (let i = 2; i < argv.length; i++) {
    const a = argv[i];
    if (!a.startsWith('--')) continue;
    const key = a.slice(2);
    if (key === 'dry-run') { out.dryRun = true; continue; }
    const val = argv[i + 1];
    i++;
    out[key] = val;
  }
  return out;
}

function requireField(obj, name) {
  if (!obj[name] || String(obj[name]).trim() === '') {
    console.error(`Missing required --${name}`);
    process.exit(2);
  }
}

async function main() {
  const args = parseArgs(process.argv);

  const apiKey = args.apikey || process.env.TELTEL_API_KEY;
  if (!apiKey && !args.dryRun) {
    console.error('Missing TELTEL_API_KEY (env) or --apikey');
    process.exit(2);
  }

  const from = (args.from || process.env.TELTEL_SMS_FROM || '').replace(/^\+/, '');
  const to = (args.to || '').trim();
  const message = args.message;
  const callback = args.callback;

  requireField({ to }, 'to');
  requireField({ message }, 'message');
  if (!from) {
    console.error('Missing --from (or TELTEL_SMS_FROM env).');
    process.exit(2);
  }

  const payload = {
    data: {
      from,
      to,
      message,
      ...(callback ? { callback } : {})
    }
  };

  const url = `${BASE_URL}/sms/text`;

  if (args.dryRun) {
    console.log(JSON.stringify({ url, payload }, null, 2));
    return;
  }

  const res = await fetch(url, {
    method: 'POST',
    headers: {
      'content-type': 'application/json',
      'x-api-key': apiKey,
    },
    body: JSON.stringify(payload),
  });

  const text = await res.text();
  let json;
  try { json = JSON.parse(text); } catch { json = null; }

  if (!res.ok) {
    console.error(`TelTel API error: HTTP ${res.status}`);
    console.error(text);
    process.exit(1);
  }

  console.log(JSON.stringify(json ?? { raw: text }, null, 2));
}

main().catch((err) => {
  console.error(err?.stack || String(err));
  process.exit(1);
});
